### ISSUE TITLE
User Role Queries: Improper Access Violation

### ISSUE THREAD

**Issue by jasonbahl** [wp-graphql/wp-graphql#890]
When querying for users, the roles should be checked before the query is made. 

For example, a public user should not be able to get a list of admin users by querying:

```
{
  users( where: { role: ADMINISTRATOR }) {
     nodes {
        id
        title
      }
  }
}
```

As that would expose who is and who is not an administrator to the public

**Comment by CodeProKid** [comment]
ahhhh, interesting 🤔

**Comment by jasonbahl** [comment]
closed by #900
